ignite/metrics/psnr.py

```diff
@@ -0,0 +1,106 @@
+from typing import Callable, Optional, Sequence, Union
+
+import torch
+import torch.nn.functional as F
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+__all__ = ["PSNR"]
+
+
+class PSNR(Metric):
+    """Computes `Peak signal-to-noise ratio (PSNR) <https://en.wikipedia.org/wiki/Peak_signal-to-noise_ratio>`_.
+
+    Args:
+        data_range (float): Range of the image. If not provided, it will be determined
```

Comment:
Let me get some info about how to deal with n-channels images and when channel do not have the same range (for example, HSV color space).